Move lots of general docs to webpage (#93)
nf-core/nf-co.re: markdown/usage/adding_own_config.md

To specify your cluster environment, add the following line to your config file:

```nextflow
process.executor = 'YOUR_SYSTEM_TYPE'
```

Many different cluster types are supported by Nextflow. For more information, please see the [Nextflow documentation](https://www.nextflow.io/docs/latest/executor.html).

Note that you may need to specify cluster options, such as a project or queue. To do so, use the `clusterOptions` config option:

```nextflow
process {
  executor = 'SLURM'
  clusterOptions = '-A myproject'
}
```
## Software Requirements
To run your selected nf-core pipeline, several software packages are required. How you satisfy these requirements is essentially up to you and depends on your system. If possible, we _highly_ recommend using either Docker or Singularity.

Please see the [`installation documentation`](usage/installation) for how to run using the below as a one-off. These instructions are about configuring a config file for repeated use.

### Docker

> Docker is a great way to run {{ cookiecutter.name }}, as it manages all software installations and allows the pipeline to be run in an identical software environment across a range of systems.

Reviewer: Might be worth doing a global search for `cookiecutter` to make sure that we removed all of these references..

Reviewer: ```suggestion
Docker is a great way to run nf-core pipelines, as it manages all software installations and allows the pipeline to be run in an identical software environment across a range of systems.
```

Nextflow has [excellent integration](https://www.nextflow.io/docs/latest/docker.html) with Docker, and beyond installing the two tools, not much else is required - nextflow will automatically fetch the Docker image that we have created and is hosted at DockerHub at run time.

To add docker support to your own config file, add the following:

```nextflow
docker.enabled = true

> process.container = "nfcore/YOUR_PIPELINE_NAME"

Reviewer: I don't think we need this line? Should already be in the base configs..

Author: True :-)

```

> Note that the DockerHub organisation name annoyingly can't have a hyphen, so is `nfcore` and not `nf-core`.

Reviewer: ```suggestion
> Note that the DockerHub organisation name annoyingly can't have a hyphen, so it's `nfcore` and not `nf-core`.
```

### Singularity image
Many HPC environments are not able to run Docker due to security issues.
[Singularity](http://singularity.lbl.gov/) is a tool designed to run on such HPC systems which is very similar to Docker.

To specify singularity usage in your pipeline config file, add the following:

```nextflow
singularity.enabled = true

> process.container = "nfcore/YOUR_PIPELINE_NAME"

Reviewer: Again, no need for this line I think, if pulling from dockerhub. Would also be good to explicitly mention that we use DockerHub also for singularity images.

```

If you intend to run the pipeline offline, nextflow will not be able to automatically download the singularity image for you. Instead, you'll have to do this yourself manually first, transfer the image file and then point to that.

First, pull the image file where you have an internet connection:

```bash

> singularity pull --name nfcore-YOURPIPELINENAME.simg nfcore/YOUR_PIPELINE_NAME

Reviewer: ```suggestion
singularity pull --name nfcore-YOURPIPELINENAME-VERSION.simg nfcore/YOURPIPELINENAME:VERSION
```

Does this need to be `docker://nfcore/YOURPIPELINENAME:VERSION`?

```

Then transfer this file and point the config file to the image:

```nextflow
singularity.enabled = true

> process.container = "/path/to/nfcore-YOURPIPELINENAME.simg"

Reviewer: ```suggestion
process.container = "/path/to/nfcore-YOURPIPELINENAME-VERSION.simg"
```

Could also be good to mention the `-with-singularity /path/to/nfcore-YOURPIPELINENAME-VERSION.simg` option.
